be/src/runtime/routine_load/kafka_consumer_pipe.h

```diff
@@ -66,10 +72,22 @@ class KafkaConsumerPipe : public StreamLoadPipe {
     }
 
     Status append_json(const char* data, size_t size, char row_delimiter) {
+        return append_json(data, size, row_delimiter, -1, -1);
+    }
+
+    Status append_json(const char* data, size_t size, char row_delimiter, int32_t partition, int64_t offset) {
+        bool need_meta = partition >= 0 && offset >= 0;
         // For efficiency reasons, simdjson requires a string with a few bytes (simdjson::SIMDJSON_PADDING) at the end.
-        ASSIGN_OR_RETURN(auto buf, ByteBuffer::allocate_with_tracker(size + simdjson::SIMDJSON_PADDING));
+        ASSIGN_OR_RETURN(auto buf, ByteBuffer::allocate_with_tracker(
+                                           size + simdjson::SIMDJSON_PADDING,
+                                           need_meta ? ByteBufferMetaType::KAFKA : ByteBufferMetaType::NONE));
         buf->put_bytes(data, size);
         buf->flip();
+        if (need_meta) {
```

Comment:
how much overhead to put the meta into the buffer meta? is it possible to always put the meta along with the buffer, so the user can assume the buffer meta is available under all conditions, that can reduce the complex of understanding the tech details.

Reply:
it's not just about overhead. This pr does not support pulsar, so meta is meaningless for pulsar